store/src/java/com/zimbra/cs/mailbox/Mailbox.java

```diff
@@ -8189,6 +8227,55 @@ public void modifyContact(OperationContext octxt, int contactId, ParsedContact p
         }
         return newAddrs;
     }
+    
+    /**
+     * Returns a list of contacts exist in emailed contacts folders.
+     * @param octxt OperationContext object
+     * @param addrs Recipient Email address list
+     * @return listOfContactsToBeModified This list consist list of Map, InternetAddress as key and Pair of ContactId and 
+     * DisplayName as Value
+     * @throws ServiceException 
+     */
+    public List<Map<InternetAddress, Pair<Integer, String>>> findContactsInEmailedContacts
+    (OperationContext octxt, Collection<Address> addrs) throws ServiceException {
+        if (addrs.isEmpty()) {
+            return Collections.emptyList();
+        }
+        List<Map<InternetAddress, Pair<Integer, String>>> list = new ArrayList<>();
+        Map<InternetAddress, Pair<Integer, String>> map = new HashMap<>();
+
+        for (Address addr : addrs) {
+            if (addr instanceof javax.mail.internet.InternetAddress) {
+                javax.mail.internet.InternetAddress iaddr = (javax.mail.internet.InternetAddress) addr;
+                try {
+                    if (!Strings.isNullOrEmpty(iaddr.getAddress())) {
+                        ZimbraQueryResults results = index.search(octxt, "inid:" + FolderConstants.ID_FOLDER_AUTO_CONTACTS + " field[email]:" + iaddr.getAddress(),
+                                EnumSet.of(MailItem.Type.CONTACT), SortBy.NONE, 1);
+                        while (results.hasNext()) {
+                            ZimbraHit hit = results.getNext();
+                            if (hit instanceof ContactHit) {
+                                ContactHit contactHit = (ContactHit) hit;
+                                Contact contact = contactHit.getContact();
+                                Map<String, String> m = contact.getFields();
+                                String fullName = m.get(ContactConstants.A_fullName);
+                                String firstName = m.get(ContactConstants.A_firstName);
+                                String emailSentName = iaddr.getPersonal();
```

Comment:
NullPointerException is thrown when a recipient (To field) does not contain display name.
i.e. To field is not  `"name" <xxx@domain.com>` but just `xxx@domain.com`

Reply:
Done, null check on emailSentName has been done. 
When user is typing address only, then Name is Null. In this case, not doing modify operation. Since there is no updation of name happened anywhere.